tests/system/python/conftest.py

```diff
@@ -285,12 +283,98 @@ def _read_data_from_pi(host, admin, password, pi_database, asset, sensor):
             return None
     return _read_data_from_pi
 
+@pytest.fixture
+def read_data_from_pi_web_api():
+    def _read_data_from_pi_web_api(host, admin, password, pi_database, af_hierarchy_list, asset, sensor):
+        """ This method reads data from pi web api """
+
+        # List of pi databases
+        dbs = None
+        # PI logical grouping of attributes and child elements
+        elements = None
+        # List of elements
+        url_elements_list = None
+        # Element's recorded data url
+        url_recorded_data = None
+        # Resources in the PI Web API are addressed by WebID, parameter used for deletion of element
+        web_id = None
+        # List of elements
+        url_elements_data_list=None
+
+        username_password = "{}:{}".format(admin, password)
+        username_password_b64 = base64.b64encode(username_password.encode('ascii')).decode("ascii")
+        headers = {'Authorization': 'Basic %s' % username_password_b64}
+
+        try:
+            conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+            conn.request("GET", '/piwebapi/assetservers', headers=headers)
+            res = conn.getresponse()
+            r = json.loads(res.read().decode())
+            dbs = r["Items"][0]["Links"]["Databases"]
+
+            if dbs is not None:
+                conn.request("GET", dbs, headers=headers)
+                res = conn.getresponse()
+                r = json.loads(res.read().decode())
+                for el in r["Items"]:
+                    if el["Name"] == pi_database:
+                        elements = el["Links"]["Elements"]
+
+            if elements is not None:
+                conn.request("GET", elements, headers=headers)
+                res = conn.getresponse()
+                r = json.loads(res.read().decode())
+                url_elements_list = r["Items"][0]["Links"]["Elements"]
```

Comment:
I would suggest to make it like:

               items = r["Items"]
               url_elements_list = None
               for el in items:
                    if el["Name"] == af_hierarchy_list[0]:
                        url_elements_list = el["Links"]["Elements"]

Reply:
And we can put a comment like, if hierarchy is foglamp/tenant1/data_piwebapi then we need to keep finding ["Links"]["Elements"] recursiverly for foglamp then tenant1, and next code block will read last node `data_piwebapi` Elements link